src/Shared/CertificateGeneration/MacOSCertificateManager.cs

```diff
@@ -301,20 +280,65 @@ protected override X509Certificate2 SaveCertificateCore(X509Certificate2 certifi
 
             if (process.ExitCode != 0)
             {
-                Log.MacOSAddCertificateToKeyChainError(process.ExitCode);
-                throw new InvalidOperationException($@"There was an error importing the certificate into the user key chain '{certificate.Thumbprint}'.
-
-{output}");
+                Log.MacOSAddCertificateToKeyChainError(process.ExitCode, output);
+                throw new InvalidOperationException("Failed to add the certificate to the keychain. Are you running in a non-interactive session perhaps?");
             }
         }
 
         Log.MacOSAddCertificateToKeyChainEnd();
-
-        return certificate;
+        return true;
```

Comment:
Does this currently return true or throw?  Perhaps this can just be void?

Reply:
@HaoK Yea, I could make it void. One of the intermediate approaches I tried was making use of the bool but it's not used anymore.